angular/projects/public-nrpti/src/app/records/utils/record-utils.ts

```diff
@@ -43,4 +45,188 @@ export class RecordUtils {
+      'Summary',
+      'Issuing Agency',
+      'Legislation Act',
+      'Regulation',
+      'Section',
+      'Sub Section',
+      'Paragraph',
+      'Description',
+      'Offence',
+      'Penalties',
+      'Site/Project',
+      'Location',
+      'Latitude',
+      'Longitude'
+    ];
+
+    let output = '';
+    output = `${csvHeaders.join(',')}\n`;
+
+    for (const row of data) {
+      let line = [];
+      line.push(escapeCsvString(row['recordName']));
+      line.push(escapeCsvString(row['recordType']));
+      line.push(moment(row['dateIssued']).format('YYYY-MM-DD'));
+
+      const issuedTo = row['issuedTo'];
+      if (issuedTo) {
+        if (issuedTo['type'] === 'Company') {
+          line.push(escapeCsvString(issuedTo['companyName']));
+        } else {
+          line.push(escapeCsvString(issuedTo['fullName']));
+        }
+      } else {
+        line.push('Unpublished');
+      }
+
+      line.push(escapeCsvString(row['summary']));
+      line.push(escapeCsvString(row['issuingAgency']));
+
+      const legislation = row['legislation'];
+
+      if (legislation) {
+        line.push(escapeCsvString(legislation['act']));
+        line.push(escapeCsvString(legislation['regulation']));
+        line.push(escapeCsvString(legislation['section']));
+        line.push(escapeCsvString(legislation['subSection']));
+        line.push(escapeCsvString(legislation['paragraph']));
+      } else {
+        line = line.concat(['', '', '', '', '', '']);
+      }
+
+      line.push(escapeCsvString(row['legislationDescription']));
+      line.push(escapeCsvString(row['offence']));
+
+      const penalties = row['penalties'];
+      let penaltiesString = '';
+      if (penalties) {
+        penaltiesString = penalties.map(penalty => new Penalty(penalty).buildPenaltyString()).join('; ');
+      }
+      line.push(escapeCsvString(penaltiesString));
+
+      line.push(escapeCsvString(row['projectName']));
+      line.push(escapeCsvString(row['location']));
+
+      const centroid = row['centroid'];
+
+      if (centroid && centroid.length === 2) {
+        line.push(centroid[1]);
+        line.push(centroid[0]);
+      } else {
+        line = line.concat(['', '']);
+      }
+
+      output += `${line.join(',')}\n`;
+    }
+
+    download(`nrced-export-${moment().format('YYYY-MM-DD')}.csv`, output);
+  }
+}
+
+/**
+ * Escape special characters in the CSV fields so the final output can be opened
+ * correcly in Excel.
+ *
+ * @static
+ * @param {any} csvField A single CSV field value
+ */
+function escapeCsvString(csvField: any): string {
+  if (!csvField) {
+    return '';
+  }
+
+  let str = csvField.toString();
+
+  // Replace newline chars
+  str = str.replace(/(?:\r\n|\r|\n)/g, '');
+
+  // Escape quotes
+  str = str.replace(/\"/g, '""');
+
+  // Escape commas
+  if (str.indexOf(',') > -1) {
+    str = `"${str}"`;
+  }
+
+  return str;
+}
+
+/**
+ * Escape special characters in the CSV fields so the final output can be opened
```

Comment:
Update doc description for download

Reply:
Fixed in https://github.com/bcgov/NRPTI/pull/798/commits/c3308888b1560b725839ee6c3de47c46c2c87768